core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1647,6 +1645,24 @@ class SnappySession(@transient private val sc: SparkContext,
 
   def setPreparedQuery(preparePhase: Boolean, paramSet: Option[ParameterValueSet]): Unit =
     sessionState.setPreparedQuery(preparePhase, paramSet)
+
+  /**
+   * Callback to execute code when a particular property is set.
+   * @param key
+   * @param value
+   */
+  def confCallback(key: String, value: String): Unit = {
+    key match {
+      case "spark.scheduler.pool" =>
+        // set the scheduler pool if the pool exists else throw an exception. 
+        if (sparkContext.getAllPools.exists(_.name == value)) {
+          sparkContext.setLocalProperty(key, value)
+        } else {
+          throw new IllegalArgumentException(s"Invalid Pool $value")
+        }
+      case _ =>
+    }
+  }
```

Comment:
> the doc also says that this property is only for JDBC clients. I am not sure about the usage of this property in Spark

Didn't get this. Weren't you just arguing above that Spark users will use the standard SparkContext.setLocalProperty way? Going by that I thought this new property handling was mostly relevant for JDBC/ODBC connections.

> I am not sure about the usage of this property in Spark because new config help doesn't mention it and older helps say that it is for users migrating from shark.

Which property you are looking at? The "spark.sql.thriftserver.scheduler.pool" is clearly the only supported way in Spark to set pool over hive-thriftserver: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L401

> I would rather go with either snappydata.scheduler.pool

Then users will have two different properties in SQLConf: "snappydata.scheduler.pool" when using SnappyContext, and "spark.sql.thriftserver.scheduler.pool" when using hive-thriftserver in smart connector mode (or when we add it in embedded mode). Whats the rationale for not using the "spark.sql.thriftserver.scheduler.pool" name that will make it consistent with existing Spark's hive-thriftserver?